Add OctoBot Script using AI article (#41)
Drakkar-Software/OctoBot-Blog: 2023-10-trading-with-ai-introduction.md

---
title: 
subtitle: 
slug: trading-with-ai-introduction
tags: ai, deep learning, trading, cryptocurrency, octobot
cover: 

> domain: blog.octobot.online

Reviewer: we would need a description and title (used in meta tag and previews)

Author: Yes

--- 

# Trading with AI for beginner's

> ## AI in Trading

Reviewer: Can we add a small intro explaining what people would get from reading this article ?

Author: Right, done

Artificial Intelligence (AI) has revolutionized how we trade. It helps in analyzing massive amounts of data, predicting market trends, and executing trades at lightning speed. To trade using AI, you need to choose a reliable AI trading software, set your trading parameters, and let the system do the rest.

![brain](https://raw.githubusercontent.com/Drakkar-Software/OctoBot-Blog/master/resources/images/trading-with-ai-introduction/trading.jpg)

## Understanding Reinforcement Learning

> Reinforcement Learning is a type of machine learning (itself a type of AI) where an agent learns to make decisions by taking actions in an environment to maximize some notion of cumulative reward. 

Reviewer: This is targeting advanced users ? (personnally I don't easily understand "agent learns to make decisions by taking actions in an environment to maximize some notion of cumulative reward")

Author: Ok I'll explain more, it targets advanced and non-advanced users

In trading, AI uses Reinforcement Learning to understand market dynamics, make accurate predictions, and execute profitable trades.

![brain](https://raw.githubusercontent.com/Drakkar-Software/OctoBot-Blog/master/resources/images/trading-with-ai-introduction/brain.jpg)

## OctoBot Script

> OctoBot Script is engineered to provide traders with a framework for crafting and testing OctoBot strategies. 

Reviewer: OctoBot Script or OctoBot script ? (we use OctoBot cloud)

Author: Oh yes it's OctoBot script

> OctoBot Script is engineered to provide traders with a framework for crafting and testing OctoBot strategies. 

Reviewer: `testing OctoBot strategies` or `testing strategies` ?

It offers a suite of keywords (Python methods) which simplifies the process of creating trades and calculating TA indicators like RSI, thus facilitating users to design their unique trading strategies. 

OctoBot Script also allows users to test their strategies using past data through the backtesting feature. With the generation of an advanced report at the end of each backtesting, users gain valuable insights into the performance of their strategies, enabling a comprehensive understanding of their effectiveness.

> ## How to use OctoPulse to trade with AI

Reviewer: OctoPulse :D

Author: Oupsie 😄

- Install OctoBot Script by following the get started guide at [pro.octobot.info](https://pro.octobot.info/)
- Install AI requirements with 
```
pip install -r requirements-rl.txt
```

> - Install the necessary to be able to run the script on your GPU by following [this tutorial](https://gretel.ai/blog/install-tensorflow-with-cuda-cdnn-and-gpu-support-in-4-easy-steps) 

Reviewer: "the necessary **dependencies** to be able" ?

- Start to train your own AI model on ETH/BTC using 
```
python3 rl.py -t -s ETH/BTC -e 10
```

> - Once done your model will be saved in the weights folder. Check this name and add it in the end of the following command to run a backtesting using your new AI model 

Reviewer: ```suggestion
- Once done your model will be saved in the weights folder. Find its name and add it in the end of the following command to run a backtesting using your new AI model 
```

```
python3 rl.py -p -s ETH/USDT -w weights/202310050722-final-dqn.h5
```
*202310050722-final-dqn.h5 is an example of weight, update it with your own*

- Here is an example output of a model built using OctoBot Script AI.

![strategy-ouput](https://raw.githubusercontent.com/Drakkar-Software/OctoBot-Blog/master/resources/images/trading-with-ai-introduction/strategy-output.png)

> If you found this content helpful, please give it a like! Your support will encourage us to create a series of detailed guides exploring more strategies and insights into AI trading.

Reviewer: we can't like in the new docs, we would need something else 😂

Author: 😄

Author: What about `If you found this content helpful, please give us feedback in our community [Discord](https://discord.com/invite/vHkcb8W) and [Telegram](https://t.me/+UisN86uSzOMKtEwy)!` ?

Reviewer: super 👍

---
title: Crypto trading using artificial intelligence
subtitle: Learn how to automate your crypto trading with AI in 5 steps
slug: trading-with-ai-introduction
tags: ai, deep learning, trading, cryptocurrency, octobot
cover: 
domain: octobot.cloud
--- 

# How to automate crypto trading using AI

Dive into the future of cryptocurrency trading using the power of AI with OctoBot script! 
We'll walk you through 5 simple steps to automate your crypto trading using artificial intelligence. 

> No matter your experience level, this guide is designed to familiarize you with essential AI terminology and provide a step-by-step process for setting up and executing your first automated cryptocurrency trade using AI.

Reviewer: `familiarize you with essential AI terminology and` is that necessary ? maybe we can remove it

Author: Yes we can, it's one of the benefits of the article but it's not the main

Author: I removed it

## AI in Trading
Artificial intelligence (AI) has revolutionized how we trade. It helps in analyzing massive amounts of data, predicting market trends, and executing trades at lightning speed. To trade using AI, you need to choose a reliable AI trading software, set your trading parameters, and let the system do the rest.

![trading](https://raw.githubusercontent.com/Drakkar-Software/OctoBot-Blog/master/resources/images/trading-with-ai-introduction/trading.jpg)

## Understanding reinforcement learning

> Reinforcement Learning is a type of machine learning (itself a type of AI) where an agent learns to make decisions by taking actions in an environment to maximize some notion of cumulative reward. An 'agent' in this context refers to the algorithm or program that is making the decisions. It operates by interacting with its environment (in this case, the trading market), taking actions (such as buying or selling stocks), and receiving rewards or penalties based on the outcome. The goal of this agent is to learn over time which actions lead to the best outcomes, in this case, the most profitable trades.

Reviewer: complicated but much clearer, thanks !
